src/Paket.Core/PaketConfigFiles/DependenciesFile.fs

```diff
@@ -262,17 +264,79 @@ type DependenciesFile(fileName,groups:Map<GroupName,DependenciesGroup>, textRepr
                     // "We don't really want Package A, but if you need it take Version X (from our resolution above)"
-                    // Maybe we can actually do this by modifying the "getVersionF" callback?
+                    // We do this we hook-in in a modified getVersionF callback which filters known packages to only contain
+                    // the version from the resolution above
+                    // Additionally we add all the requirements, but with locked versions.
+                    let getVersionFromFirstResolution sources strategy groupName packageName =
+                        let resolvedVersion =
+                            match resolved |> Map.tryFind packageName with
+                            | Some res -> Some res.Version
+                            | _ -> None
+                        // getVersionF is already cached by upper layers so that's propably fine?
+                        getVersionF sources strategy groupName packageName
+                        |> Seq.filter (fun (ver, _) ->
+                            match resolvedVersion with Some v -> v = ver | None -> true)
+
+                    tracefn  "Trying to find a valid resolution considering runtime dependencies..."
+                    let runtimeResolutionDeps =
+                        resolved
+                        |> Map.toSeq
+                        |> Seq.map snd
+                        |> Seq.map (fun p ->
+                            let oldDepsInfo = step1Deps |> Seq.tryFind (fun d -> d.Name = p.Name)
+                            { Name = p.Name
+                              VersionRequirement = VersionRequirement (VersionRange.Exactly p.Version.AsString, PreReleaseStatus.All)
+                              // How to get that for the package?
+                              ResolverStrategyForDirectDependencies =
+                                match oldDepsInfo with
+                                | Some d -> d.ResolverStrategyForDirectDependencies
+                                | None -> group.Options.ResolverStrategyForDirectDependencies
+                              ResolverStrategyForTransitives =
+                                match oldDepsInfo with
+                                | Some d -> d.ResolverStrategyForTransitives
+                                | None -> group.Options.ResolverStrategyForTransitives
+                              Parent = PackageRequirementSource.DependenciesFile "runtimeresolution.dependencies"
+                              Graph = []
+                              Sources = group.Sources
+                              Settings =
+                                match oldDepsInfo with
+                                | Some d -> d.Settings
+                                | None -> p.Settings })
+                        |> fun des -> Seq.append des runtimeDeps
+                        |> Seq.distinctBy (fun p -> p.Name) |> Set.ofSeq
```

Comment:
This part seems to be quite slow as it takes quite a while after `Trying to find a valid resolution considering runtime dependencies...` has been printed until the second resolution starts...